components/butterfly/src/server.rs

```diff
@@ -232,7 +232,7 @@ pub struct Server {
     member_id: Arc<String>,
     // TODO (CM): This is currently public because butterfly tests
     // depends on it being so. Refactor so it can be private.
-    pub member:               Arc<RwLock<Myself>>,
+    pub member:               Arc<Lock<Myself>>,
```

Comment:
Another thing that *might* be worth doing is pulling the locking code into the `Myself` type and putting that into a submodule (I'd suggest `sync`) as I did with [`GatewayState`](https://github.com/habitat-sh/habitat/pull/6937). That way you can force _all_ access to go through well-defined interfaces. Currently, there's a number of places where there is direct access outside `Server` since submodules can still directly access non-`pub` fields of supermodules.

The value of eliminating direct access is that it becomes harder to ignore the lock ordering concerns. If a consumer can directly access the lock; we're relying 100% on the author of the code to remember to update all its callers with appropriate lock annotations.

Reply:
Ok, I'll take a look at that.